src/targets/node/nodeLauncher.ts

```diff
@@ -92,4 +96,37 @@ export class NodeLauncher extends NodeLauncherBase<INodeLaunchConfiguration> {
 
     doLaunch(this.restarters.create(runData.params));
   }
+
+  /**
+   * @inheritdoc
+   */
+  protected createLifecycle(
+    cdp: Cdp.Api,
+    run: IRunData<INodeLaunchConfiguration>,
+    { targetId }: Cdp.Target.TargetInfo,
+  ): INodeTargetLifecycleHooks {
+    return {
+      initialized: async () => {
+        if (run.params.stopOnEntry) {
+          const params = {
+            url: absolutePathToFileUrl(resolve(run.params.cwd, run.params.program)),
+            lineNumber: 0,
+            columnNumber: 0,
+          };
+
+          await cdp.Debugger.setBreakpointByUrl(params);
```

Comment:
If stopOnEntry is set, we just want to stop where we are paused from `--inspect-brk`, ie don't autoresume. We don't need to set a breakpoint.

Reply:
That was the first thing I tried, after dissecting the existing node-debug2 extension. The issue is that that ends up breaking inside our bootloader.js, rather than the user's code, since that's the first thing that gets required.